test/version_sorter_test.rb

```diff
@@ -59,18 +73,49 @@ def test_does_not_raise_on_number_overflow
 
   def test_handles_non_version_data
     non_versions = [
-      "", " ", ".", "-", "ćevapčići", "The Quick Brown Fox", '!@#$%^&*()',
+      " ", ".", "-", "", "ćevapčići", "The Quick Brown Fox", '!@#$%^&*()',
       "<--------->", "a12a8a4a22122d01541b62193e9bdad7f5eda552", "1." * 65
     ]
     sorted = [
-      "<--------->", "-", "The Quick Brown Fox",
-      "a12a8a4a22122d01541b62193e9bdad7f5eda552", "ćevapčići",
-      "", " ", ".", '!@#$%^&*()', "1." * 65
+      "", " ", '!@#$%^&*()', "-", ".", "<--------->",
+      "The Quick Brown Fox", "a12a8a4a22122d01541b62193e9bdad7f5eda552",
+      "ćevapčići", "1." * 65
```

Comment:
I think the trouble of the current sorting mechanism for non-versions can be summed up as:

- It will [basically] sort as expected for alphanumeric characters
- But all other characters (or lack thereof), they are effectively ignored, and the output order will be more based on input order

This is why I added the `strcmp` usage specifically for comparisons with 0 comparison chunks, which will make sorting for those special cases deterministic.